.github/workflows/int.yml

```diff
@@ -11,35 +11,38 @@ env:
   GOPRIVATE: github.com/step-security
 jobs:
   integration-test:
+    if: github.event.repository.fork == false
     permissions:
       contents: read
     runs-on: ubuntu-latest
     steps:
-      - uses: step-security/harden-runner@v2
+      - uses: step-security/harden-runner@c6295a65d1254861815972266d5933fd6e532bdf # v2.11.1
         with:
           egress-policy: audit
       - name: Checkout
-        uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - name: Set up Go
-        uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4
+        uses: actions/setup-go@0aaccfd150d50ccaeb58ebd88d36e91967a5f35b # v5.4.0
         with:
           go-version: 1.24.1
           
       - name: Configure .netrc
+        env:
+          PAT: ${{ secrets.PAT }}
         run: |
           if [[ ! -e "~/.netrc" ]]; then
               touch ~/.netrc
           fi
-          printf "machine github.com login stepsecurity-infra-bot password ${{ secrets.PAT }}" >>~/.netrc
```

Comment:
Best practice is to not use `${{ ... }}` inside a `run:` statement and instead favor env variables.